### ISSUE TITLE
Ingress

### ISSUE THREAD

**Issue by pierreozoux** [xetys/hetzner-kube#15]
Do you want this cluster to have a default ingress with TLS and Let's encrypt?

I think it would be a good idea to offer a default option.

I didn't manage yesterday to set it up properly (was tired, and kube-lego is no longer the recommended way, only dns method works in traefik, and I didn't manage to make it work in a reasonable amount of time.)

Here looks like a good article about how to set it properly with:
 - nginx ingress
 - cert manager
 - Let's encrypt

**Comment by xetys** [comment]
Can you tell more about that "kube-lego is no longer recommended"...I'm using it in production and don't have any problems with that.

Generally, I was thinking of a `hetzner-kube cluster addons install|remove|list` command with the offering of different tools, such as helm or ingress.

**Comment by pierreozoux** [comment]
About lego:
https://github.com/kubernetes/charts/tree/master/stable/kube-lego

**Comment by JohnnyQQQQ** [comment]
I think this is one of the most important features to have, otherwise it's not very simple to throw some traffic at the cluster if you have more than one edge-router, as one would need to setup a bunch of (ha)proxies using a floating ip to throw the traffic at the edge-routers.

**Comment by Baughn** [comment]
As a novice trying to make some use of Kubernetes...

Well, I'd say this is critical. Hetzner-kube mostly works now, but although I'm able to start applications and have them run, I've yet to figure out how to get ingress working. None of the tutorials seem to cover everything needed, and the documentation is too vast--I've tried reading it, but it's difficult to know what matters and what doesn't.

I probably wouldn't see any need to change a reasonable default setup, but I can't even get started without having a functional base to start from.

**Comment by xetys** [comment]
## tl;dr

```
# helm install --name ingress --set rbac.create=true,controller.kind=DaemonSet,controller.service.type=ClusterIP,controller.hostNetwork=true stable/nginx-ingress
```

This will install an ingress controller in the way I always describe. As `DaemonSet`, so it runs on all nodes  (you can limit that with nodeSelector to `role=edge`) and with `hostNetwork=true` so you actually can reach the nginx pod. If you have 2 workers, you just can add those IPs as A-Record to a domain and create ingress resources as you want. For test purposes, you can add xip.io-Domains. 

## Addon

The same command is now available by `hetzner-kube cluster addon install ingress -n my-cluster` (and requires helm installed)

closed by https://github.com/xetys/hetzner-kube/commit/39270bae0d1b5fef511c5cd6139522b61e987299

**Comment by exocode** [comment]
I am a quite newbie on kubernetes.

I have troubles to get my apps reaching out to the www with the ingress controller by exposing/connecting to a floating ip.

Do I have to add your options `rbac.create=true,controller.kind=DaemonSet,controller.service.type=ClusterIP,controller.hostNetwork=true` when I install the controller through hetzner-kube?
I am asking because I cannot find these options in the source code (https://github.com/xetys/hetzner-kube/blob/master/pkg/addons/addon_hcloud_controller_manager.go)

How to pass these options with the `hetzner-kube` command?

My goal: 
`my.domain.com -> Hetzner FLOATING IP -> NODE(s) -> MyDemoNginxPod`

**Comment by JohnnyQQQQ** [comment]
@exocode it's in the ingress file 

https://github.com/xetys/hetzner-kube/blob/master/pkg/addons/addon_ingress.go

**Comment by JohnnyQQQQ** [comment]
> I have troubles to get my apps reaching out to the www with the ingress controller by exposing/connecting to a floating ip.

Did you add the floating ip it to your network interface?

**Comment by exocode** [comment]
Hi @JohnnyQQQQ, thank you for responding my questions.

You mean by adding `sudo ip addr add 123.456.789.0 dev eth0`? Yes I did that. I can see the attached IP when I use `ip addr`.

What me makes confusing - _because the whole stuff is new for me_ - is, that I dont know which service to choose to expose my apps: `LoadBalancer` or `NodePort`? In my understanding Hetzner has no loadbalancer, so I have to "install the ingress plugin".

I also saw someone provided a short snippet to achieve automatic assignment of floating ips if a node dies: https://github.com/xetys/hetzner-kube/issues/58#issuecomment-375089121 
Is that snippet then used to be included in a DaemonSet?
Wouldn't it be good to add that as a hetzner-kube plugin or a helm chart?

Sorry for making noise, but it's very hard for me to get through kubernetes. I am regularly developing apps, I am not a devOp but familiar with docker stuff.
